File: src/main/java/org/jcodec/containers/mp4/boxes/TextConfigBox.java
package org.jcodec.containers.mp4.boxes;

import java.nio.ByteBuffer;

import org.jcodec.common.io.NIOUtils;
import org.jcodec.platform.Platform;

/**
 * This class is part of JCodec ( www.jcodec.org ) This software is distributed
 * under FreeBSD License
 * 
 * @author The JCodec project
 * 
 */
public class TextConfigBox extends FullBox {

    private String textConfig;

    public TextConfigBox(Header atom) {
        super(atom);
    }

    public static String fourcc() {
        return "txtC";
    }

    public static TextConfigBox createTextConfigBox(String textConfig) {
        TextConfigBox box = new TextConfigBox(new Header(fourcc()));
        box.textConfig = textConfig;
        return box;
    }

    @Override
    public void parse(ByteBuffer input) {
        super.parse(input);
        textConfig = NIOUtils.readNullTermStringCharset(input, Platform.UTF_8);
    }

    @Override
    protected void doWrite(ByteBuffer out) {
        super.doWrite(out);
        NIOUtils.writeNullTermString(out, textConfig);
    }

    @Override
    public int estimateSize() {
        return 13 + Platform.getBytesForCharset(textConfig, Platform.UTF_8).length;
    }

    public String getTextConfig() {
        return textConfig;
    }
}

File: src/main/java/org/jcodec/containers/mp4/boxes/TextConfigBox.kt
package org.jcodec.containers.mp4.boxes

import org.jcodec.common.io.NIOUtils
import org.jcodec.platform.Platform
import java.nio.ByteBuffer

/**
 * This class is part of JCodec ( www.jcodec.org ) This software is distributed
 * under FreeBSD License
 *
 * @author The JCodec project
 */
class TextConfigBox(atom: Header) : FullBox(atom) {
    var textConfig: String? = null
        private set

    override fun parse(input: ByteBuffer) {
        super.parse(input)
        textConfig = NIOUtils.readNullTermStringCharset(input, Platform.UTF_8)
    }

    override fun doWrite(out: ByteBuffer) {
        super.doWrite(out)
        NIOUtils.writeNullTermString(out, textConfig)
    }

    override fun estimateSize(): Int {
        return 13 + Platform.getBytesForCharset(textConfig, Platform.UTF_8).size
    }

    companion object {
        @JvmStatic
        fun fourcc(): String {
            return "txtC"
        }

        fun createTextConfigBox(textConfig: String?): TextConfigBox {
            val box = TextConfigBox(Header(fourcc()))
            box.textConfig = textConfig
            return box
        }
    }
}